src/manager.cpp

```diff
@@ -17,12 +21,52 @@ Manager::Manager(const std::shared_ptr<boost::asio::io_context>& ioCon,
 
         // Set up minimal things that is needed before bus name is claimed.
         m_worker->performInitialSetup();
+
+        // set async timer to detect system VPD collection status. This is
+        // required so that bus claim and event loop is not blocked while all
+        // FRU VPD is being collected.
+        setTimerForSystemVPDDetection();
 #endif
+        // TBD: register interface(s) for exposed APIs here.
     }
     catch (const std::exception& e)
     {
         logging::logMessage("VPD-Manager service failed.");
         throw;
     }
 }
+
+#ifdef IBM_SYSTEM
+void Manager::setTimerForSystemVPDDetection()
+{
+    static boost::asio::steady_timer timer(*m_ioContext);
+
+    // timer for 2 seconds
+    auto asyncCancelled = timer.expires_after(std::chrono::seconds(2));
+
+    (asyncCancelled == 0) ? std::cout << "Timer started" << std::endl
+                          : std::cout << "Timer re-started" << std::endl;
+
+    timer.async_wait([this](const boost::system::error_code& ec) {
+        if (ec == boost::asio::error::operation_aborted)
```

Comment:
It is in case any handler is already in execution queue due to timer expiration and expires_after is called again at that time. Then those handlers will not get cancelled automatically as it happens when the handler is not in execution queue and will any how execute with error code set.